lib/src/main/java/com/orange/ods/compose/component/button/OdsIconToggleButtonsRow.kt

```diff
@@ -0,0 +1,169 @@
+import androidx.compose.ui.unit.dp
+import com.orange.ods.compose.component.OdsComponentApi
+import com.orange.ods.compose.component.utilities.DisabledInteractionSource
+import com.orange.ods.compose.component.utilities.Preview
+import com.orange.ods.compose.component.utilities.UiModePreviews
+import com.orange.ods.compose.theme.OdsDisplaySurface
+import com.orange.ods.compose.theme.OdsTheme
+
+/**
+ * <a href="https://system.design.orange.com/0c1af118d/p/06a393-buttons/b/79b091" target="_blank">ODS Buttons</a>.
+ *
+ * A group of toggle buttons. Only one option in a group of toggle buttons can be selected and active at a time.
+ * Selecting one option deselects any other.
+ *
+ * @param iconToggleButtons Contains the buttons to display in the toggle group
+ * @param selectedIndex The [iconToggleButtons] list index of the selected button.
+ * @param onSelectedIndexChange Callback to be invoked when the selection change.
+ * @param modifier optional [Modifier] for this IconToggleButton
+ * @param displaySurface optional allow to force the button display on a dark or light
+ * surface. By default the appearance applied is based on the system night mode value.
+ */
+@Composable
+@OdsComponentApi
+fun OdsIconToggleButtonsRow(
+    iconToggleButtons: List<OdsIconToggleButtonsRowItem>,
+    selectedIndex: Int,
+    onSelectedIndexChange: (Int) -> Unit,
+    modifier: Modifier = Modifier,
+    displaySurface: OdsDisplaySurface = OdsDisplaySurface.Default
+) {
+    Row(
+        modifier = modifier
+            .height(IntrinsicSize.Min)
+            .selectableGroup()
+            .border(
+                width = 1.dp,
+                color = buttonToggleBorderColor(displaySurface)
+            )
+    ) {
+        iconToggleButtons.forEachIndexed { index, iconToggleButton ->
+            IconToggleButtonsRowItem(
+                index = index,
+                iconToggleButton = iconToggleButton,
+                selected = selectedIndex == index,
+                displaySurface = displaySurface
+            ) { clickedButtonIndex ->
+                onSelectedIndexChange(clickedButtonIndex)
+            }
+            if (index < iconToggleButtons.size) {
+                Divider(
+                    modifier = Modifier
+                        .fillMaxHeight()
+                        .width(1.dp),
+                    color = buttonToggleBorderColor(displaySurface)
+                )
+            }
+        }
+    }
+}
+
+data class OdsIconToggleButtonsRowItem(
+    val icon: Painter,
+    val iconDescription: String,
+    val enabled: Boolean = true
+)
+
+@Composable
+private fun IconToggleButtonsRowItem(
+    index: Int,
+    iconToggleButton: OdsIconToggleButtonsRowItem,
+    selected: Boolean,
+    displaySurface: OdsDisplaySurface,
+    onClick: (Int) -> Unit
+) {
+    val iconTint by animateColorAsState(buttonToggleIconColor(displaySurface, selected))
+    val backgroundAlpha by animateFloatAsState(if (selected) 0.12f else 0f)
+
+    Box(
```

Comment:
What is the purpose of the box ? Maybe I missed something but it seems that the following code works the same:

```kotlin
@Composable
private fun IconToggleButtonsRowItem(
    index: Int,
    iconToggleButton: OdsIconToggleButtonsRowItem,
    selected: Boolean,
    displaySurface: OdsDisplaySurface,
    onClick: (Int) -> Unit
) {
    val iconTint by animateColorAsState(buttonToggleIconColor(displaySurface, selected))
    val backgroundAlpha by animateFloatAsState(if (selected) 0.12f else 0f)

    Icon(
        modifier = Modifier
            .background(color = buttonToggleBackgroundColor(displaySurface).copy(alpha = backgroundAlpha))
            .padding(12.dp)
            .clickable(interactionSource = remember { DisabledInteractionSource() }, indication = null) { onClick(index) },
        painter = iconToggleButton.icon,
        contentDescription = iconToggleButton.iconDescription,
        tint = iconTint
    )
}
```

In fact this is a bit strange that it is still working with `DisabledInteractionSource`.

Reply:
The `DisabledInteractionSource` is to remove ripple effect when clicking on the icon.